uast/ann/ann.go

```diff
@@ -134,28 +181,33 @@ func (r *Rule) Do(actions ...Action) *Rule {
 
 // HasInternalType matches a node if its internal type matches the given one.
 func HasInternalType(it string) Predicate {
-	return func(n *uast.Node) bool {
-		if n == nil {
-			return false
-		}
-
-		return n.InternalType == it
+	return &predicate{
+		eval: func(n *uast.Node) bool {
+			if n == nil {
+				return false
+			}
+			return n.InternalType == it
+		},
+		desc: fmt.Sprintf("@InternalType='%s'", it),
```

Comment:
I also expected this to be decoupled. The type would be `hasInternalTypePredicate` instead of `predicate`, so it would be possible to test for the type from a separate function to determine the representation.